library_information_validator.py

```diff
@@ -0,0 +1,318 @@
+# load libraries
+from __future__ import print_function
+import argparse
+from openpyxl import load_workbook
+import coloredlogs
+import logging
+import re
+import couchdb
+import numbers
+import decimal
+
+
+
+
+#global variable
+WARNINGS = 0
+
+
+
+
+# Set up a logger with colored output
+logger = logging.getLogger(__name__)
+logger.propagate = False  # Otherwise the messages appeared twice
+coloredlogs.install(level='INFO', logger=logger,
+                    fmt='%(asctime)s %(levelname)s %(message)s')
+
+
+
+
+class ProjectSheet:
+#Class Attributes
+    SHEET_NAME = 'Sample information'
+    FIRST_LINE = 20  # First line where user submitted data is located
+    SAMPLE_NAME_COL = 'O'
+    A_RATIO_COL = 'R'  # A260:A280 ratio
+    CONC_COL = 'S'
+    VOL_COL = 'T'
+    RIN_COL = 'V'
+    SAMPLE_TYPE = 'O8'
+    PLATE_ID = 'M6'
+    PROJECT_NAME_USER_SHEET = 'M3'
+    sample_recommendation_sheet ='/Users/franziska.franziska/Documents/FB_NGI/bioinfo/standalone_scripts_git/standalone_scripts/Samplesheet_converter/Sample_requirements.xlsx'
+
+# Initializer / Instance attributes
+    def __init__(self, sample_info_sheet):
+        self.sample_info_sheet = sample_info_sheet
+
+# instance methods
+    def getAccessUserSheet(self):
+        wb = load_workbook(self.sample_info_sheet, read_only=True, data_only=True)
+        ws = wb[ProjectSheet.SHEET_NAME]
```

Comment:
If you save this sheet as a variable here and before loading it checking if it is loaded: 
```
if self.work_sheet is None:
    wb = load_workbook ...
    ws = wb[...
    self.work_sheet = ws
return self.work_sheet
```
Then you won't have to read it from the file every time this method is called. This requires you add `self.work_sheet = None` to the `__init__` function above as well.